### ISSUE TITLE
decoder(encoder(in)) != in (specific case and suggestion)

### ISSUE THREAD

**Issue by x94carbone** [aspuru-guzik-group/selfies#9]
I have found a very specific case in which it appears the decoder applied to an encoded canonical SMILES string does not yield the same input. That said, it *is* actually the same molecule, so I am not sure if this not unintended, so to speak. Here is the code to reproduce.

```
in_smiles = 'Cc1nnc(C2CC23CCN(c2ncccc2Br)CC3)o1'
x = selfies.encoder(in_smiles)
out_smiles = selfies.decoder(x)

In: in_smiles
Out: 'Cc1nnc(C2CC23CCN(c2ncccc2Br)CC3)o1'

In: out_smiles
Out: 'Cc1nnc(C2CC22CCN(c3ncccc3Br)CC2)o1'
```

Note the subtle difference. However, explicitly canonicalizing them fixes the issue.

```
from rdkit import Chem
a = Chem.CanonSmiles(out_smiles)
b = Chem.CanonSmiles(in_smiles)
a == b == in_smiles  # True
```

where

```
In: a
Out: 'Cc1nnc(C2CC23CCN(c2ncccc2Br)CC3)o1'
```

I do not know enough about how this encoding works to suggest an explicit solution. However, if you're willing to include `rdkit` dependencies, this does seem to address the ambiguity! I think further testing would be required to be sure.

**Comment by MarioKrenn6240** [comment]
Thanks x94carbone, that is very helpful. I will look into it and fix it in a follow-up version. It's a nice fix that you found with producing canonical SMILES from RDKit, which also helps identifying the issue.

**Comment by x94carbone** [comment]
Happy to help @MarioKrenn6240. Yeah, I had a hunch that they were too similar to be completely wrong and figured that perhaps `rdkit` had a method for mapping two different strings that corresponded to the same structure to some consistent result. Let me know if there's any other way I can assist!

**Comment by MarioKrenn6240** [comment]
This issue got fixed in SELFIES v1.0.0. Thank you!
